tools/chplcheck/src/rules.py

```diff
@@ -370,7 +471,13 @@ def is_range_like(node: AstNode):
             if not is_range_like(exprs[0]):
                 continue
 
-            yield iterand, loop
+            s = "\n".join(range_to_text(exprs[0].location(), lines))
```

Comment:
`range_to_text` is only called without a `\n.join` once. This makes me think there should be a single `range_to_lines` function (used in the one case where you do need to join not-by-newline), and `range_to_text` should itself return a string.